buildenv/jenkins/common/build.groovy

```diff
@@ -565,36 +565,51 @@ def add_node_to_description() {
     currentBuild.description = TMP_DESC + "<a href=${JENKINS_URL}computer/${NODE_NAME}>${NODE_NAME}</a>"
 }
 
+def cleanWorkspace (KEEP_WORKSPACE) {
+        if (KEEP_WORKSPACE) {
+		try {	
+			cleanWs disableDeferredWipeout: true, deleteDirs: true
+		} catch (Exception e) {
+			try {
+				if (SPEC.contains('win')) {
+					WORKSPACE = sh(script: "cygpath -u '${env.WORKSPACE}'", returnStdout: true).trim()
+				}
+				if (WORKSPACE) {
+					sh "rm -fr ${WORKSPACE}/*"
+				}
+			} catch(Exception f) {
+				println("Exception: ${f}")
```

Comment:
Just to be clear, this will allow the build to not fail if the cleanups both fail.

Reply:
Do we want to fail/abort if cleanup cannot be performed?  Guessing that would be preferrable to troubleshooting a failure later in a clone or compile.

Reply:
I think from Dev perspective, not failing is preferred. Particularly at the end of a build. But that workflow is what got us into this state. I'm fine with not failing, just wanted to make sure it was clear.